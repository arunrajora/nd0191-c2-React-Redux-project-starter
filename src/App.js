import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Route, Routes } from 'react-router-dom';
import './App.css';
import AddQuestion from './components/AddQuestion';
import Home from './components/Home';
import Leaderboard from './components/Leaderboard';
import Login from './components/Login';
import NavBar from './components/NavBar';
import ProtectedRoute from './components/ProtectedRoute';
import QuestionDetails from './components/QuestionDetails';
import { loadInitialData } from './store/actions';
import LoadingBar from 'react-redux-loading-bar';

function App() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(loadInitialData());
  }, [dispatch]);
  return (
    <div>
      <LoadingBar />
      <NavBar />
      <Routes>
        <Route
          index
          element={
            <ProtectedRoute>
              <Home />
            </ProtectedRoute>
          }
        />
        <Route path='/login' element={<Login />} />
        <Route
          path='/add'
          element={
            <ProtectedRoute>
              <AddQuestion />
            </ProtectedRoute>
          }
        />
        <Route
          path='/leaderboard'
          element={
            <ProtectedRoute>
              <Leaderboard />
            </ProtectedRoute>
          }
        />
        <Route
          path='/questions/:question_id'
          element={
            <ProtectedRoute>
              <QuestionDetails />
            </ProtectedRoute>
          }
        />
        <Route
          path='*'
          element={
            <ProtectedRoute>
              <h1 align='center'>404 page</h1>
            </ProtectedRoute>
          }
        />
      </Routes>
    </div>
  );
}

export default App;
