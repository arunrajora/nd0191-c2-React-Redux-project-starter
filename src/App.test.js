import React from 'react';
import { render } from './test-utils';
import App from './App';

test('renders App and matches previous snapshot', () => {
  const { container } = render(<App />);
  expect(container.innerHTML).toMatchSnapshot();
});
