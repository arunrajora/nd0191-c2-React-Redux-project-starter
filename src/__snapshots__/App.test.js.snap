// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App and matches previous snapshot 1`] = `"<div><div style=\\"direction: ltr;\\"><div style=\\"width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;\\" class=\\"\\"></div><div style=\\"display: table; clear: both;\\"></div></div><header class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root\\"><div class=\\"MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root\\"><div class=\\"MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root\\"><div class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1083xa6-MuiTypography-root\\">Employee Poll</div><div class=\\"MuiBox-root css-1lvtzne\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"account of current user\\" aria-controls=\\"menu-appbar\\" aria-haspopup=\\"true\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"MenuIcon\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-h7cdpm-MuiTypography-root\\">Employee Poll</div><div class=\\"css-t96um6-MuiStack-root\\"><a tabindex=\\"0\\" class=\\"MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root\\" style=\\"\\" href=\\"/\\">Home<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root\\" href=\\"/add\\">Add<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root\\" href=\\"/leaderboard\\">Leaderboard<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a></div></div></div></header><div><h1 align=\\"center\\">LOGIN</h1><form><div class=\\"css-c5xtse-MuiStack-root\\"><div class=\\"MuiFormControl-root MuiTextField-root css-v3jsul-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"outlined-basic\\" id=\\"outlined-basic-label\\">userId</label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"outlined-basic\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-173wfxe\\"><span>userId</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root css-1z0lssb-MuiFormControl-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"outlined-adornment-password\\">Password</label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" autocomplete=\\"password\\" id=\\"outlined-adornment-password\\" type=\\"password\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"toggle password visibility\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"VisibilityIcon\\"><path d=\\"M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-173wfxe\\"><span>Password</span></legend></fieldset></div></div><button class=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"-1\\" type=\\"submit\\" disabled=\\"\\">Submit</button></div></form></div></div>"`;
